### ISSUE TITLE
IndexError: list index out of range while running setup.py

### ISSUE THREAD

**Issue by gijzelaerr** [ska-sa/montblanc#100]
```
gijs@bloempot:~/Work/montblanc (master)
λ  python setup.py register
NVIDIA cub installed: False
NVIDIA cub archive downloaded: False
Traceback (most recent call last):
  File "setup.py", line 142, in <module>
    install_cub()
  File "setup.py", line 111, in install_cub
    dl_cub(cub_url, cub_zip_file)
  File "setup.py", line 35, in dl_cub
    remote_file_size = int(meta.getheaders("Content-Length")[0])
IndexError: list index out of range
```

**Comment by sjperkins** [comment]
Looks like HTTP request failed. Probably need to support resumable download as this also causes issues on the Jenkins test server.

**Comment by gijzelaerr** [comment]
personally I truly dislike this installation method, it fails more than it works.

**Comment by sjperkins** [comment]
Reasonably fixed by #111. HTTP Server isn't guaranteed to supply `Content-Length` header, in which case one has to trust that the server will close the connection when all data has been sent.
